Address WireMock transitive dependencies vulnerabilities

wzieba opened the pull request:
This PR addresses security vulnerabilities added by transitive dependencies of WireMock.

## To Test:

Green light on CI should be enough.

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-Android/milestone/291">25.4</a>. This milestone is due in less than 4 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

wzieba committed:
Constraint WireMock dependencies

So it uses dependencies without reported security vulnerabilities

wzieba committed:
Update configuration for `wiremock` to not duplicate constraints and setup in both modules

Use `runtimeOnly` instead of `constraints` as the later works only if dependency is already in the classpath.

wzieba committed:
Bring back `wiremockHttpClientVersion`

As this is replacment dependency, maybe it's worth to keep a version extracted? I'm not sure, so I revert this change to keep things unchanged

wzieba committed:
Extract WireMock transitive dependencies security message

To not duplicate strings

ParaskP7 commented on libs/mocks/build.gradle:
```diff
@@ -18,11 +18,32 @@ android {
+            because("newer versions of WireMock use Java APIs not available on Android")
+        }
+
+        def wireMockSecurity = "version shipped with WireMock 2.26.3 contains security vulnerabilities"
```
Praise (❤️): Nice de-duplication touch here!

ParaskP7 commented on libs/mocks/build.gradle:
```diff
@@ -18,11 +18,32 @@ android {
 
 dependencies {
-    implementation("com.github.tomakehurst:wiremock:$wiremockVersion") {
+    api("com.github.tomakehurst:wiremock") {
```
Question (❓): Why the `api` change, just so that you could maybe remove the configuration from the main `WordPress` module? 🤔

1. I also saw that the dependency analysis report was suggesting the `api` change, so I am comfortable with it, I just wanted to understand the need, just as an FYI for me.
2. Btw, now that you made this `api` change, and removed the `wiremock` dependency from the main `WordPress` module, the dependency analysis report is asking us to actually declare this transitive dependency directly. If you're curious, just run `./gradlew buildHealth` on this branch and check-it-out for yourself.

wzieba replied:
> Why the api change, just so that you could maybe remove the configuration from the main WordPress module? 🤔

Yes

I understand it'd be nice to add wiremock as direct dependency in `WordPress`, but unfortunately it's not that easy with constraints - they won't be applied in `WordPress` module.

btw: I've tried to run `buildHealth` task, but I was getting Java's heap memory exception. Do you experience similar locally?

ParaskP7 replied:
> I understand it'd be nice to add wiremock as direct dependency in WordPress, but unfortunately it's not that easy with constraints - they won't be applied in WordPress module.

Ok, got it, let's leave it at that for now. 👍 

> btw: I've tried to run buildHealth task, but I was getting Java's heap memory exception. Do you experience similar locally?

Have you applies the new configuration from secrets lately? 😊 

Meaning, do you have the `dependency.analysis.android.ignored.variants=xyz` configuration on your local `gradle.properties` file to exclude all variants and keep just one, there are lots, maybe that's why you get the memory exception! 😅

wzieba replied:
ahh thanks!

ParaskP7 reviewed (changes requested):
👋 @wzieba !

I have reviewed this PR and everything is looking good, thanks for dealing with this problem on yet another such repo! 🌟 

-----

I have left a question (❓) for us to discuss. I am **NOT** going to approve this PR just yet to give us some time to discuss first. Also, I have an extra warning (⚠️) comment (see below).

-----

`EXTRAS`
1. Warning (⚠️): Wouldn't we want to ignore `wiremock`'s dependencies with the Dependabot configuration file, just like we did on all other repos? 🤔

wzieba committed:
Update Dependabot config to ignore WireMock's transitive dependencies
